--!strict
local DoublePendulumComponent = require(script.Parent.Parent.DoublePendulumComponent)
local React = require(game:GetService("ReplicatedStorage").Packages.React)

describe("DoublePendulumComponent", function()
    it("should exist", function()
        expect(DoublePendulumComponent).to.be.ok()
    end)

    it("should instantiate with default props", function()
        local instance = DoublePendulumComponent.new()
        expect(instance).to.be.ok()
        expect(instance.props.origin).to.equal(Vector3.new(0, 20, 0))
        expect(instance.props.length1).to.equal(10)
        expect(instance.props.length2).to.equal(10)
        expect(instance.props.mass1).to.equal(1)
        expect(instance.props.mass2).to.equal(1)
        expect(instance.props.angle1).to.equal(math.pi / 2)
        expect(instance.props.angle2).to.equal(math.pi / 2)
        expect(instance.props.color1).to.equal(Color3.fromRGB(255, 21, 21))
        expect(instance.props.color2).to.equal(Color3.fromRGB(255, 21, 21))
        expect(instance.props.color3).to.equal(Color3.fromRGB(0, 200, 255))
        expect(instance.props.CanCollide).to.equal(true)
        expect(instance.props.rodThickness_x).to.equal(1)
        expect(instance.props.rodThickness_z).to.equal(0.5)
        expect(instance.props.rod1_Transparency).to.equal(0.9)
        expect(instance.props.rod2_Transparency).to.equal(0.9)
        expect(instance.props.ballRadius).to.equal(2)
        expect(instance.props.ballRadiusTrail).to.equal(instance.props.ballRadius * 0.7)
        expect(instance.props.ball0_Transparency).to.equal(0.3)
        expect(instance.props.ball1_Transparency).to.equal(0.3)
        expect(instance.props.ball2_Transparency).to.equal(0)
    end)
end)
